src/hd/pci.c

```diff
@@ -1927,5 +1931,94 @@ void hd_read_sdio(hd_data_t *hd_data)
   free_str_list(sf_bus);
 }
 
+
+/*
+ * nd (nvdimm)
+ */
+void hd_read_nd(hd_data_t *hd_data)
+{
+  int i, blk_cnt = 0;
+  hd_t *hd, *hd2;
+  str_list_t *sf_bus, *sf_bus_e;
+  char *sf_dev, *drv, *drv_name, *modalias, *s, *t;
+
+  sf_bus = read_dir("/sys/bus/nd/devices", 'l');
+
+  if(!sf_bus) {
+    ADD2LOG("sysfs: no such bus: nd\n");
+    return;
+  }
+
+  for(sf_bus_e = sf_bus; sf_bus_e; sf_bus_e = sf_bus_e->next) {
+    sf_dev = new_str(hd_read_sysfs_link("/sys/bus/nd/devices", sf_bus_e->str));
+
+    ADD2LOG(
+      "  nd device: name = %s\n    path = %s\n",
+      sf_bus_e->str,
+      hd_sysfs_id(sf_dev)
+    );
+
+    drv_name = NULL;
+    drv = new_str(hd_read_sysfs_link(sf_dev, "driver"));
+    if(drv) {
+      drv_name = strrchr(drv, '/');
+      if(drv_name) drv_name++;
+    }
+
+    ADD2LOG("    driver = \"%s\"\n", drv_name);
+
+    if((modalias = get_sysfs_attr_by_path(sf_dev, "modalias"))) {
+      modalias = canon_str(modalias, strlen(modalias));
+      ADD2LOG("    modalias = \"%s\"\n", modalias);
+    }
+
+    if(hd_read_sysfs_link(sf_dev, "block")) {
+      ADD2LOG("    block device\n");
+
+      char *sysfs_id = new_str(hd_sysfs_id(sf_dev));
+      // go 3 levels up (bus, region, namespace)
+      for(i = 0; i < 3; i++) {
+        if((t = strrchr(sysfs_id, '/'))) *t = 0;
+      }
+
+      // if we haven't created it yet...
+      if(!hd_find_sysfs_id(hd_data, sysfs_id)) {
+        hd = add_hd_entry(hd_data, __LINE__, 0);
+
+        hd->base_class.id = bc_storage;
+        hd->sub_class.id = sc_sto_other;
```

Comment:
Just a question: would it make sense to have a new separate subclass for the NVDIMM devices? How can e.g. YaST find out that a particular devices is an NVDIMM device?